google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcConversions.java

```diff
@@ -845,6 +860,61 @@ private BlobInfo blobInfoDecode(Object from) {
     return toBuilder.build();
   }
 
+  private com.google.iam.v1.Policy policyEncode(Policy from) {
+    com.google.iam.v1.Policy.Builder to = com.google.iam.v1.Policy.newBuilder();
+    ifNonNull(from.getEtag(), ByteString::copyFromUtf8, to::setEtag);
+    ifNonNull(from.getVersion(), to::setVersion);
+    from.getBindingsList().stream().map(bindingCodec::encode).forEach(to::addBindings);
+    return to.build();
+  }
+
+  private Policy policyDecode(com.google.iam.v1.Policy from) {
+    Policy.Builder to = Policy.newBuilder();
+    if (!from.getEtag().isEmpty()) {
+      to.setEtag(from.getEtag().toStringUtf8());
+    }
+    to.setVersion(from.getVersion());
+    ImmutableList<Binding> bindings =
+        from.getBindingsList().stream()
+            .map(bindingCodec::decode)
+            .collect(ImmutableList.toImmutableList());
+    to.setBindings(bindings);
+    return to.build();
+  }
+
+  private com.google.iam.v1.Binding bindingEncode(Binding from) {
+    com.google.iam.v1.Binding.Builder to = com.google.iam.v1.Binding.newBuilder();
+    to.setRole(from.getRole());
+    to.addAllMembers(from.getMembers());
+    to.setCondition(iamConditionCodec.encode(from.getCondition()));
+    return to.build();
+  }
+
+  private Binding bindingDecode(com.google.iam.v1.Binding from) {
+    Binding.Builder to = Binding.newBuilder();
+    to.setRole(from.getRole());
+    to.setMembers(from.getMembersList());
+    to.setCondition(iamConditionCodec.decode(from.getCondition()));
+    return to.build();
+  }
+
+  private Expr conditionEncode(Condition from) {
+    Expr.Builder to = Expr.newBuilder();
+    to.setExpression(from.getExpression());
+    to.setTitle(from.getTitle());
+    to.setDescription(from.getDescription());
+    // apiary doesn't have a "location" field like grpc does
```

Comment:
so we don't want the location field in GRPC either?

Reply:
At this point, I don't think so. GCS is using a generic representation here and I don't think they'll populate location for grpc when there is no place to populate it for apiary.